src/server/routes/weatherLocation.js

```diff
@@ -0,0 +1,219 @@
+
+function formatWeatherLocationForResponse(item) {
+	return {
+		id: item.id,
+		identifier: item.identifier,
+		gps: item.gps,
+		note: item.note
+	};
+}
+
+/**
+ * Route for getting all weather locations.
+ */
+router.get('/', optionalAuthMiddleware, async (req, res) => {
+	const conn = getConnection();
+	try {
+		const rows = await WeatherLocation.getAll(conn);
+		res.json(rows.map(formatWeatherLocationForResponse));
+	} catch (err) {
+		log.error(`Error while performing GET weather location details query: ${err}`, err);
+		failure(res, 500, `Error while retrieving weather locations ${err}`);
+	}
+});
+
+// This checks params for both edit and create. In principle they could differ since not all needed for create
+// but they are the same due to current routing for now.
+function validateWeatherLocationParams(params) {
+	const validParams = {
+		type: 'object',
+		required: ['identifier', 'gps'],
+		properties: {
+			// Removed id from properties list since it is set to undefined no matter what is passed.
+			identifier: {
+				type: 'string',
+				minLength: 1
+			},
+			gps: {
+				oneOf: [
+					{
+						type: 'object',
+						required: ['latitude', 'longitude'],
+						properties: {
+							latitude: { type: 'number', minimum: '-90', maximum: '90' },
+							longitude: { type: 'number', minimum: '-180', maximum: '180' }
+						}
+					},
+					{ type: 'null' }
+				]
+			},
+			note: {
+				oneOf: [
+					{ type: 'string' },
+					{ type: 'null' }
+				]
+			}
+		}
+	}
+	const paramsValidationResult = validate(params, validParams);
+	return { valid: paramsValidationResult.valid, errors: paramsValidationResult.errors };
+}
+
+/**
+ * Fetches and inserts weather data for a newly created weather location.
+ * If the location has no existing weather data, it retrieves data starting from the earliest reading timestamp.
+ * @param location - Instance of WeatherLocation.
+ * @param conn - The database connection.
+ */
+async function addWeatherDataForLocation(location, conn) {
+	// Find last timestamp for that location 
+	let earliestDate = await WeatherData.getLatestTimeStamp(location.id, conn);
+
+	// If no weather data exists, use earliest reading timestamp
+	if (earliestDate === null) {
+		earliestDate = await Reading.getEarliestTimeStamp(conn);
+		if (earliestDate === null) {
+			log.warn('addWeatherDataForLocation(): Cannot find earliest time from WeatherData or Reading.');
+			return;
+		}
+	}
+
+	// this only gets the data for the current day(?)
+	// earliestDate = earliestDate.format('YYYY-MM-DD');
+	// const latestDate = moment().format('YYYY-MM-DD');
+	earliestDate = '2025-12-11'; // test date
+	const latestDate = moment().format('YYYY-MM-DD');
+
+	// Fetch weather data for this location's coordinates
+	const weatherData = await fetchWeatherData(
+		location.gps.latitude,
+		location.gps.longitude,
+		earliestDate,
+		latestDate
+	);
+
+	// Insert weather data records
+	if (weatherData) {
+		await conn.tx(async t => {
```

Comment:
Should this be done via a new model function in weatherData so it is generally available?